x-pack/plugins/secops/public/containers/kpi_network/index.tsx

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { getOr } from 'lodash/fp';
+import React from 'react';
+import { Query } from 'react-apollo';
+
+import { GetKpiNetworkQuery, KpiNetworkData } from '../../graphql/types';
+import { createFilter } from '../helpers';
+import { QueryTemplateProps } from '../query_template';
+import { kpiNetworkQuery } from './index.gql_query';
+
+export interface KpiNetworkArgs {
+  id: string;
+  kpiNetwork: KpiNetworkData;
+  loading: boolean;
+}
+
+export interface KpiNetworkProps extends QueryTemplateProps {
+  children: (args: KpiNetworkArgs) => React.ReactNode;
+}
+
+export class KpiNetworkQuery extends React.PureComponent<KpiNetworkProps> {
+  public render() {
+    const {
+      id = 'kpiNetworkQuery',
+      children,
+      filterQuery,
+      sourceId,
+      startDate,
+      endDate,
+    } = this.props;
+
+    return (
+      <Query<GetKpiNetworkQuery.Query, GetKpiNetworkQuery.Variables>
+        query={kpiNetworkQuery}
+        fetchPolicy="cache-and-network"
+        notifyOnNetworkStatusChange
+        variables={{
+          sourceId,
+          timerange: {
+            interval: '12h',
+            from: startDate!,
+            to: endDate!,
+          },
+          filterQuery: createFilter(filterQuery),
+        }}
+      >
+        {({ data, loading }) => {
+          const kpiNetwork = getOr({}, `source.KpiNetwork`, data);
+          return children({
+            id,
+            loading,
+            kpiNetwork,
+          });
+        }}
+      </Query>
+    );
+  }
+}
```

Comment:
Great code! Really clean and easy to read through.

Since this has only `props` an optional but interesting thing I believe you can do is change this out for a `pure` from `recompose` which will make it easer later when we eventually use `reactmemo`:
https://reactjs.org/docs/react-api.html#reactmemo

Since we're backporting to `7.1` I don't think we can use `reactmemo` just yet, but using recompose pure is a good step.

The code would be even smaller with it. It would something like this with an implicit return:

```ts
export const KpiNetworkQuery = pure<{ children }> => (
    <Query<GetKpiNetworkQuery.Query, GetKpiNetworkQuery.Variables>
        query={kpiNetworkQuery}
        fetchPolicy="cache-and-network"
        notifyOnNetworkStatusChange
        variables={{
          sourceId,
          timerange: {
            interval: '12h',
            from: startDate!,
            to: endDate!,
          },
          filterQuery: createFilter(filterQuery),
        }}
      >
        {({ data, loading }) => {
          const kpiNetwork = getOr({}, `source.KpiNetwork`, data);
          return children({
            id,
            loading,
            kpiNetwork,
          });
        }}
      </Query>)
```

That's not exact syntax up there so you have to fiddle a bit, and this isn't required, but overall for the eventual evolution towards hooks and react functions we are trying to keep what we can within `pure` recompose. That and it's less code to write :-)

Let me know what you think.

Reply:
ah, I got it. There is no specific reason that I have to make `KpiNetworkQuery` a class, agree to `pure` from `recompose` here.